lightning/src/util/config.rs

```diff
@@ -126,6 +126,29 @@ pub struct ChannelHandshakeConfig {
 	///
 	/// [`KeysInterface::get_shutdown_scriptpubkey`]: crate::chain::keysinterface::KeysInterface::get_shutdown_scriptpubkey
 	pub commit_upfront_shutdown_pubkey: bool,
+
+	/// `their_channel_reserve_proportion_millionth` is proportion of channel_value
+	/// to configure counterparty's channel_reserve i.e `their_channel_reserve_satoshis`
+	/// for both outbound and inbound channels.
+	///
+	/// `their_channel_reserve_satoshis` is the minimum amount that the other node is to keep as a direct payment.
+	/// This ensures that if our counterparty broadcasts a revoked state, we can punish them by
+	/// claiming at least this value on chain.
+	///
+	/// channel_reserve values greater than 30% could be considered highly unreasonable, since that
+	/// amount can never be used for payments.
+	/// Also, if our selected channel_reserve for counterparty and counterparty's selected channel_reserve
+	/// for us, sum upto or greater than channel_value, channel negotiations will fail.
+	///
+	/// Note: Versions of LDK earlier than v0.0.104 will fail to read channels with any channel_reserve
+	/// different than default value.
+	///
+	/// Default value: is 1% of channel_value, configured as 10,000 millionth
+	/// Minimum value: If calculated proportional value is < 1000 sats, will be treated as 1000 sats instead,
+	/// 				1000 sats is just a safe implementation-specific lower bound.
+	/// Maximum value: 1000,000, any values larger than 1Million will be treated as 1Million (or 100%) instead,
+	/// although channel negotiations will probably fail in that case.
```

Comment:
Nit:
```suggestion
	/// although channel negotiations will fail in that case.
```

It's guaranteed to fail as the `channel_reserve` is required to be set to the at least the dust limit by both parties.